### PR TITLE
Allow create_dataset to take a Type and Dataspace, Fix #1084

### LINKED ISSUE #1083: `maxdims` or `max_dims` in `dataspace()`
In `dataspaces.jl`, the documentation has `dataspace(dims::Tuple; maxdims::Tuple=dims)` while the actual code has `max_dims` as the keyword argument.

### LINKED ISSUE #1084: Using `Type` and `Dataspace` in `create_dataset()`
According to the documentation of `create_dataset(parent, path, datatype, dataspace; properties...)`, `datatype` can be either `Datatype` or `Type`, and `dataspace` can be either `Dataspace` or `Dims`. However, it throws an error if one uses `datatype::Type` with `dataspace::Dataspace`.

### PR DESCRIPTION
**Pull request by mkitti**
- Fix #1083 maxdims -> max_dims in dataspace documentation
- Fix #1084, create_dataset with Type and Dataspace

### CODE REVIEW

**Comment by mkitti** [comment]
This is a bug fix and it's been a week, so I'm just going to merge this.

**Comment by musm** [comment]
LGTM

File: src/dataspaces.jl [hunk 1]
```diff
@@ -78,13 +78,17 @@ dataspace(n::Nothing) = Dataspace(API.h5s_create(API.H5S_NULL))
...
 
 Construct a simple `Dataspace` for the given dimensions `dims`. The maximum
 dimensions `maxdims` specifies the maximum possible size: `-1` can be used to
 indicate unlimited dimensions.
 """
 dataspace(sz::Dims{N}; max_dims::Union{Dims{N},Tuple{}}=()) where {N} =
     _dataspace(sz, max_dims)
+dataspace(sz::Dims{N}, max_dims::Union{Dims{N},Tuple{}}) where {N} =
+    _dataspace(sz, max_dims)
+dataspace(dims::Tuple{Dims{N},Dims{N}}) where {N} = _dataspace(first(dims), last(dims))
```
Comment (musm) [inline src/dataspaces.jl:91]: should we document these additional methods? These  were added for additional flexibility?

Comment (mkitti) [reply src/dataspaces.jl:91]: I think that we just implemented what wa already documented per #1084, but yes the documentation should reflect the new methods.

Comment (musm) [reply src/dataspaces.jl:91]: I see the fix for adding the methods for `create_dataset` as per documentation, I was just curious on the new `dataspace` methods.

**Pull request merged**